hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieClusteringConfig.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.config;
+
+import org.apache.hudi.common.config.DefaultHoodieConfig;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * Clustering specific configs.
+ */
+public class HoodieClusteringConfig extends DefaultHoodieConfig {
+
+  public static final String SCHEDULE_CLUSTERING_STRATEGY_CLASS = "hoodie.clustering.schedule.strategy.class";
+  public static final String DEFAULT_SCHEDULE_CLUSTERING_STRATEGY_CLASS =
+      "org.apache.hudi.client.clustering.schedule.SparkBoundedDayBasedScheduleClusteringStrategy";
+
+  public static final String RUN_CLUSTERING_STRATEGY_CLASS = "hoodie.clustering.run.strategy.class";
+  public static final String DEFAULT_RUN_CLUSTERING_STRATEGY_CLASS =
+      "org.apache.hudi.client.clustering.run.SparkBulkInsertBasedRunClusteringStrategy";
+
+  // Turn on inline clustering - after few commits a inline compaction will be run
+  public static final String INLINE_CLUSTERING_PROP = "hoodie.clustering.inline";
+  private static final String DEFAULT_INLINE_CLUSTERING = "false";
+
+  public static final String INLINE_CLUSTERING_NUM_COMMIT_PROP = "hoodie.clustering.inline.num.commits";
+  private static final String DEFAULT_INLINE_CLUSTERING_NUM_COMMITS = "4";
+
+  public static final String CLUSTERING_TARGET_PARTITIONS = "hoodie.clustering.target.partitions";
+  public static final String DEFAULT_CLUSTERING_TARGET_PARTITIONS = String.valueOf(2);
```

Comment:
hello , i am interesting  about the scenario to limit the partition num of clustering. does cluster all partitions have some problem?

Reply:
I think we do file listing per partition in the first strategy implementation. So if you have lot of partitions, that can cause some degradation. I reached out on slack to help with testing. I sent you commands to run inline clustering. Will you be able to test this PR by setting this config to a large value?

Reply:
1. "So if you have lot of partitions, that can cause some degradation" will resolved in file listing  https://issues.apache.org/jira/browse/HUDI-1292?
2. i am happy to test this scenario

Reply:
Yes, that ticket can resolve issues to some extent. But, in some cases, there are large day based tables with 1000s of partitions. we probably dont want to scan all partitions in those cases. So i think its useful to have this config.

Reply:
@lw309637554 let me know if you got a chance to test this scenario (or clustering in general)